### DISCUSSION TITLE
InstantiationError: org.jboss.resteasy.spi.ResteasyDeployment

### DISCUSSION THREAD

**Discussion by marcosgopen** [resteasy/resteasy#3126] [Q&A] [answered]
In our project we have these dependencies:
 resteasy-jaxrs-3.0.23.Final
 resteasy-core-spi-4.7.2.Final
Both have org.jboss.resteasy.spi.ResteasyDeployment, in core-spi as an Interface, in jaxrs as a Class.
Could I ask you why they have same name and namespace?
Thanks in advance!

**Comment by marcosgopen** [comment, marked as answer]
The reason is these are two different majors. More info here:
https://developer.jboss.org/en/resteasy/blog/2019/03/28/resteasy-4-is-coming-soon
"the big resteasy-jaxrs and resteasy-client modules have been split into resteasy-core-spi, resteasy-client-api, resteasy-core and resteasy-client, with the first and second ones to be considered as public modules, for which we're expected to retain backward compatibility till next major release."
Thanks @xstefank for the help.

**Comment by jamezp** [comment]
Yes. You definitely do not want to mix versions like that.
